### PR TITLE
update ci workflows to run weekly

### PR DESCRIPTION
**Pull request by cqc-melf**

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -32,31 +35,31 @@ jobs:
...
     - run: git fetch --depth=1 origin +refs/tags/*:refs/tags/* +refs/heads/*:refs/remotes/origin/*
     - name: Set up Python 3.9
-      if: github.event_name == 'pull_request' || github.event_name == 'release' || contains(github.ref, 'refs/heads/wheel') || github.event_name == 'push'
+      if: github.event_name == 'pull_request' || github.event_name == 'release' || contains(github.ref, 'refs/heads/wheel') || github.event_name == 'schedule'
       uses: actions/setup-python@v3
       with:
         python-version: '3.9'
     - name: Build and test including remote checks (3.9) mypy
-      if:  (matrix.os == 'macos-11') && ((github.event_name == 'pull_request' &&  github.event.pull_request.head.repo.full_name == github.repository) || github.event_name == 'release' || contains(github.ref, 'refs/heads/wheel') || github.event_name == 'push')
+      if:  (matrix.os == 'macos-11') && ((github.event_name == 'pull_request' &&  github.event.pull_request.head.repo.full_name == github.repository) || github.event_name == 'release' || contains(github.ref, 'refs/heads/wheel') || github.event_name == 'push'  || github.event_name == 'schedule')
```
Comment (cqc-melf) [inline .github/workflows/build_and_test.yml:43]: I am not sure why this check is run on `push`, too. (other then for all other extensions...

**Comment by cqc-melf** [comment]
Should be a very simmilar PR like for pytket-aqt, see one comment about the `push`

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (cqc-alec) [reply .github/workflows/build_and_test.yml:43]: This is legacy, because it used to only work on 3.9.

**Review by cqc-alec — APPROVED** [review]

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (cqc-melf) [reply .github/workflows/build_and_test.yml:43]: Will make a note to update this in the future.

**Pull request merged**